extensions/hp-extension/src/index.tsx

```diff
@@ -13,6 +14,19 @@ export default {
    */
   id,
 
+  /**
+   * Register the hanging protocol handlers
+   * @param param0 
+   */
+  preRegistration: ({
+    servicesManager,
+  }) => {
+    const { HangingProtocolService } = servicesManager.services;
+    HangingProtocolService.addCustomViewportSetting('zoomPan', "Set initial zoom, WITH camera event", setViewportZoomPan);
```

Comment:
Folder/file organization suggestion only:
I would recommend to isolate those 3 methods into a config file.
Also the config file and viewCodeAttribute, setViewportZoomPan, lateralityAttribute should have same parent

Reply:
Eventually I agree, but for now leaving them as is is ok.